include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -474,7 +474,7 @@ namespace CUDAWrappers
 
     template <int dim, typename Number, typename functor>
     __global__ void
-    apply_kernel_shmem(const functor &                              func,
+    apply_kernel_shmem(functor                                      func,
```

Comment:
What's the reason for this change?

Reply:
The current function works fine if the functor doesn't have any data but if there is any data you won't be able to access it. The reason is that this function is called when you launch a kernel, if you pass the functor by reference the pointers to the data won't be visible from the GPU. If you copy the functor, the pointers are copied to the GPU and everything works fine.